go/vt/vtadmin/cluster/cluster.go

```diff
@@ -768,6 +768,116 @@ func (c *Cluster) getShardSets(ctx context.Context, keyspaces []string, keyspace
 	return shardsByKeyspace, nil
 }
 
+// GetCellInfos returns a list of ClusterCellInfo objects for cells in the
+// given cluster.
+//
+// If req.Cells is set, cells are restricted only to cells with those names.
+// Note: specifying a cell name that does not exist in the cluster fails the
+// overall request.
+//
+// If req.NamesOnly is set, each ClusterCellInfo will only contain the Cluster
+// and Name fields. req.Cells takes precedence over this option.
+func (c *Cluster) GetCellInfos(ctx context.Context, req *vtadminpb.GetCellInfosRequest) ([]*vtadminpb.ClusterCellInfo, error) {
```

Comment:
Obligatory question of "do we need tests for either of these"

Reply:
yes and yes

Reply:
annnd done